bundles/org.openhab.core.thing/src/test/java/org/openhab/core/thing/firmware/FirmwareEventFactoryTest.java

```diff
@@ -59,19 +60,20 @@ public void testCreateEventForUnknownType() throws Exception {
 
     @Test
     public void testNullStatusInfo() {
-        assertThrows(IllegalArgumentException.class, () -> FirmwareEventFactory.createFirmwareStatusInfoEvent(null));
+        assertThrows(IllegalArgumentException.class,
+                () -> FirmwareEventFactory.createFirmwareStatusInfoEvent(nullFirmwareStatusInfo));
     }
 
     @Test
     public void testNullUpdateProgressInfo() {
         assertThrows(IllegalArgumentException.class,
-                () -> FirmwareEventFactory.createFirmwareUpdateProgressInfoEvent(null));
+                () -> FirmwareEventFactory.createFirmwareUpdateProgressInfoEvent(nullFirmwareUpdateProgressInfo));
     }
 
     @Test
     public void testNullUpdateResultInfo() {
         assertThrows(IllegalArgumentException.class,
-                () -> FirmwareEventFactory.createFirmwareUpdateResultInfoEvent(null));
+                () -> FirmwareEventFactory.createFirmwareUpdateResultInfoEvent(nullFirmwareUpdateResultInfo));
     }
```

Comment:
If these workaround are to fix compilations errors because null is not allowed, I think it would be better to just remove these tests. AFAIK that's what we do in other code. We also remove checks to validate arguments for null once they are annotated to be non-null.

Reply:
I removed them.